### PR TITLE
Underarmour dev

### PR DESCRIPTION
**Pull request by Adrien-B**

### CODE REVIEW

**Comment by yujingwu** [comment]
@davidB  should someone handle this pull request?

File: sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    def "reject Notification without NotificationId"() {
+  		when:
+  		def req = new RequestDef("https://foo/cloudconnector/dt00/thirdpartynotification")
+  		def res = sut.onNotification(ctx, req)
+  		then:
+  		res.isBad()
+    }
+*/
+
+	def "accept userId subscripe"() {
```
Comment (davidB) [inline sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy]: typo: subscribe

File: sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+		def info = new DeviceInfo("", Option.apply(null), null, "", Option.apply(null))
+		def resp = new Response(HttpURLConnection.HTTP_OK, "application/json", '''{"id":42}''')
+		def res =sut.onSubscribeResponse(ctx, req, info, resp)
+
+		then:
+		res.isGood()
+		res.get() == Option.apply(info.withExtId("42"))
+	}
+
+	def "accept webhook subscripe"() {
```
Comment (davidB) [inline sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy]: typo: subscribe

File: sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+		when:
+		def hookJs = ["callback_url": ctx.parameters()['notificationCallback'] ,
+					  "shared_secret": ctx.parameters()['sharedSecret'] ,
+					  "subscription_type": "application.actigraphies"]
```
Comment (davidB) [inline sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy]: those parameters are null/undefined in context

File: sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+	}
+
+	def "fetch actigraphies with metrics"() {
+		when:
+		def ts = "2015-03-01T23:12:20.687025+00:00"
+		def msg = readFile(this, "actigraphiesExemple.json")
+		def req = new RequestDef(null).addQueryParams(["underArmourTs":ts])
+		def fetchedResponse = new Response(HttpURLConnection.HTTP_OK, "application/json", '''
+					'''+ msg + '''
+			}''')
```
Comment (davidB) [inline sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy]: Could t be simplified into `new Response(HttpURLConnection.HTTP_OK, "application/json", msg)` ?

File: sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+		res.get()[3] == expectedEvents[3]
+		res.get()[4] == expectedEvents[4]
+		res.get()[5] == expectedEvents[5]
+		res.get()[6] == expectedEvents[6]
+		res.get()[7] == expectedEvents[7]
+		res.get()[8] == expectedEvents[8]
+		res.get()[9] == expectedEvents[9]
+		res.get()[10] == expectedEvents[10]
+		res.get().size() == expectedEvents.size()
+		res.get() == expectedEvents
```
Comment (davidB) [inline sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy]: every `res.get()[i] == expectedEvents[i]` could be in to a loop (including 0) and I guess could be removed as res.get() == expectedEvents check the content of list

File: sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+		]
+
+		then:
+		res.isGood()
+		res.get()[1] == expectedEvents[1]
+		res.get()[2] == expectedEvents[2]
+		res.get()[3] == expectedEvents[3]
+		res.get()[4] == expectedEvents[4]
+		res.get().size() == expectedEvents.size()
+		res.get() == expectedEvents
```
Comment (davidB) [inline sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy]: same comment as above

File: sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy (hunk 5 above)
Comment (Adrien-B) [inline sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy]: yes but It's easier to fix the failing test that way.

File: sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy (hunk 6 above)
Comment (Adrien-B) [inline sample-underamour/src/test/groovy/io/samsungsami/underarmour/MyCloudConnectorSpec.groovy]: same

**Pull request merged**